src/o365/spo/commands/site/site-add.ts

```diff
@@ -192,61 +217,342 @@ class SpoSiteAddCommand extends SpoCommand {
+          };
+
+          return request.post(requestOptions);
+        })
+        .then((res: string): void => {
+          const json: ClientSvcResponse = JSON.parse(res);
+          const response: ClientSvcResponseContents = json[0];
+          if (response.ErrorInfo) {
+            reject(response.ErrorInfo.ErrorMessage);
+          }
+          else {
+            const operation: SpoOperation = json[json.length - 1];
+            let isComplete: boolean = operation.IsComplete;
+            if (!wait || isComplete) {
+              resolve();
+              return;
+            }
+
+            setTimeout(() => {
+              this.waitUntilFinished(JSON.stringify(operation._ObjectIdentity_), this.spoAdminUrl as string, resolve, reject, cmd, this.context as FormDigestInfo, this.dots, this.timeout);
+            }, operation.PollingInterval);
+          }
+        });
+    });
+  }
+
   public options(): CommandOption[] {
     const options: CommandOption[] = [
       {
         option: '--type [type]',
-        description: 'Type of modern sites to list. Allowed values TeamSite|CommunicationSite, default TeamSite',
-        autocomplete: ['TeamSite', 'CommunicationSite']
+        description: 'Type of modern sites to list. Allowed values TeamSite|CommunicationSite|ClassicSite, default TeamSite',
+        autocomplete: ['TeamSite', 'CommunicationSite', 'ClassicSite']
       },
       {
-        option: '-u, --url <url>',
-        description: 'Site URL (applies only to communication sites)'
+        option: '-t, --title <title>',
+        description: 'Site title'
       },
       {
         option: '-a, --alias <alias>',
-        description: 'Site alias, used in the URL and in the team site group e-mail (applies only to team sites)'
+        description: 'Site alias, used in the URL and in the team site group e-mail (applies to type TeamSite)'
       },
       {
-        option: '-t, --title <title>',
-        description: 'Site title'
+        option: '-u, --url <url>',
+        description: 'Site URL (applies to type CommunicationSite, ClassicSite)'
       },
       {
-        option: '-d, --description [description]',
-        description: 'Site description'
+        option: '--owner <owner>',
```

Comment:
I'm afraid that having `owner` and `owners` will be confusing. What if instead we only had `owners` and see if we can set multiple owners on classic site. If that's not possible, then we'd show validation error when type is set to `ClassicSite` and multiple owners specified

Reply:
Added a check for the ClassicSite to ensure just one owner is passed in the owners parameter. Haven't found a way to pass multiple owners to the site creation endpoint